### ISSUE TITLE
ForEach-Object Parallel Splat with ScriptBlock unable to access $using variables in pwsh 7.2.5

### ISSUE THREAD

**Issue by kahnman** [PowerShell/PowerShell#17848]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

When I try to run the following code using pwsh 7.2.5...
``` powershell 
$myNumberString = 'My Number: '
$myArray = @(1,2,3,4,5,6,7)
$foreachScriptBlock = {
    Write-Host "$($using:myNumberString) $_"
}
$foreachSplat = @{}
$foreachSplat.Add('Parallel',$foreachScriptBlock)
$myArray | ForEach-Object @foreachSplat
```

I get the errors:
``` powershell
InvalidOperation:
Line |
   2 |      Write-Host "$($using:myNumberString) $_"
     |                    ~~~~~~~~~~~~~~~~~~~~~
     | A Using variable cannot be retrieved. A Using variable can be used only with Invoke-Command, Start-Job, or InlineScript in the script workflow. When it is used with Invoke-Command, the Using variable is valid only if the script block is invoked on a remote computer.
 1
InvalidOperation:
Line |
   2 |      Write-Host "$($using:myNumberString) $_"
     |                    ~~~~~~~~~~~~~~~~~~~~~
     | A Using variable cannot be retrieved. A Using variable can be used only with Invoke-Command, Start-Job, or InlineScript in the script workflow. When it is used with Invoke-Command, the Using variable is valid only if the script block is invoked on a remote computer.
InvalidOperation:
Line |
   2 |      Write-Host "$($using:myNumberString) $_"
     |                    ~~~~~~~~~~~~~~~~~~~~~
     | A Using variable cannot be retrieved. A Using variable can be used only with Invoke-Command, Start-Job, or InlineScript in the script workflow. When it is used with Invoke-Command, the Using variable is valid only if the script block is invoked on a remote computer.
 3
 2
InvalidOperation:
Line |
   2 |      Write-Host "$($using:myNumberString) $_"
     |                    ~~~~~~~~~~~~~~~~~~~~~
     | A Using variable cannot be retrieved. A Using variable can be used only with Invoke-Command, Start-Job, or InlineScript in the script workflow. When it is used with Invoke-Command, the Using variable is valid only if the script block is invoked on a remote computer.
InvalidOperation:
Line |
   2 |      Write-Host "$($using:myNumberString) $_"
     |                    ~~~~~~~~~~~~~~~~~~~~~
     | A Using variable cannot be retrieved. A Using variable can be used only with Invoke-Command, Start-Job, or InlineScript in the script workflow. When it is used with Invoke-Command, the Using variable is valid only if the script block is invoked on a remote computer.
 5
 4
InvalidOperation:
Line |
   2 |      Write-Host "$($using:myNumberString) $_"
     |                    ~~~~~~~~~~~~~~~~~~~~~
     | A Using variable cannot be retrieved. A Using variable can be used only with Invoke-Command, Start-Job, or InlineScript in the script workflow. When it is used with Invoke-Command, the Using variable is valid only if the script block is invoked on a remote computer.
 6
InvalidOperation:
Line |
   2 |      Write-Host "$($using:myNumberString) $_"
     |                    ~~~~~~~~~~~~~~~~~~~~~
     | A Using variable cannot be retrieved. A Using variable can be used only with Invoke-Command, Start-Job, or InlineScript in the script workflow. When it is used with Invoke-Command, the Using variable is valid only if the script block is invoked on a remote computer.
 7
```

Note that the above code works when run using pwsh 7.1.1.

### Expected behavior

```console
> $myNumberString = 'My Number: '
> $myArray = @(1,2,3,4,5,6,7)
> $foreachScriptBlock = {
>>     Write-Host "$($using:myNumberString) $_"
>> }
> $foreachSplat = @{}
> $foreachSplat.Add('Parallel',$foreachScriptBlock)
$myArray | ForEach-Object @foreachSplat
My Number:  1
My Number:  2
My Number:  5
My Number:  6
My Number:  3
My Number:  7
My Number:  4
```

### Actual behavior

```console
The "My Number: " string is not written, only the numbers from $myArray. See the errors above.
```

### Error details

```console
> get-error

Exception             :
    Type        : System.Management.Automation.RuntimeException
    ErrorRecord :
        Exception             :
            Type    : System.Management.Automation.ParentContainsErrorRecordException
            Message : A Using variable cannot be retrieved. A Using variable can be used only with Invoke-Command,
Start-Job, or InlineScript in the script workflow. When it is used with Invoke-Command, the Using variable is valid
only if the script block is invoked on a remote computer.
            HResult : -2146233087
        CategoryInfo          : InvalidOperation: (:) [], ParentContainsErrorRecordException
        FullyQualifiedErrorId : UsingWithoutInvokeCommand
        InvocationInfo        :
            ScriptLineNumber : 2
            OffsetInLine     : 19
            HistoryId        : -1
            Line             :     Write-Host "$($using:myNumberString) $_"

            PositionMessage  : At line:2 char:19
                               +     Write-Host "$($using:myNumberString) $_"
                               +                   ~~~~~~~~~~~~~~~~~~~~~
            CommandOrigin    : Internal
        ScriptStackTrace      : at <ScriptBlock>, <No file>: line 2
    TargetSite  :
        Name          : GetUsingValue
        DeclaringType : System.Management.Automation.VariableOps, System.Management.Automation, Version=7.2.5.500,
Culture=neutral, PublicKeyToken=31bf3856ad364e35
        MemberType    : Method
        Module        : System.Management.Automation.dll
    Message     : A Using variable cannot be retrieved. A Using variable can be used only with Invoke-Command,
Start-Job, or InlineScript in the script workflow. When it is used with Invoke-Command, the Using variable is valid
only if the script block is invoked on a remote computer.
    Data        : System.Collections.ListDictionaryInternal
    Source      : System.Management.Automation
    HResult     : -2146233087
    StackTrace  :
   at System.Management.Automation.VariableOps.GetUsingValue(MutableTuple tuple, String usingExpressionKey, Int32
index, ExecutionContext context)
   at System.Management.Automation.Interpreter.FuncCallInstruction`5.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
CategoryInfo          : InvalidOperation: (:) [], RuntimeException
FullyQualifiedErrorId : UsingWithoutInvokeCommand
InvocationInfo        :
    ScriptLineNumber : 2
    OffsetInLine     : 19
    HistoryId        : -1
    Line             :     Write-Host "$($using:myNumberString) $_"

    PositionMessage  : At line:2 char:19
                       +     Write-Host "$($using:myNumberString) $_"
                       +                   ~~~~~~~~~~~~~~~~~~~~~
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 2
```

### Environment data

```powershell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by 237dmitry** [comment]
Your example works without `$using`

```powershell
$myNumberString = 'My Number: '
$myArray = @(1,2,3,4,5,6,7)

# $foreachScriptBlock = [scriptblock]::Create( "Write-Host $myNumberString `$_" )
$foreachScriptBlock = { "Write-Host $myNumberString $_" }

$foreachSplat = @{}
$foreachSplat.Add('Parallel',$foreachScriptBlock)

$myArray | ForEach-Object @foreachSplat
```

**Comment by jborean93** [comment]
This is a duplicate of https://github.com/PowerShell/PowerShell/issues/12378 which has been fixed in 7.3.x and I believe they are trying to backport the fix to the 7.2.x series but not sure on the status of that.

**Comment by kasini3000** [comment]
from i tested，ps 7.2 ps6.2 has this problem.
but ps 7.0,ps7.1 normal

------
```powershell
#but this code normal for all ps6,ps7
$myNumberString = 'My Number: '
$myArray = @(1,2,3,4,5,6,7)

$myArray | ForEach-Object -Parallel {
	Write-Host "$($using:myNumberString) $_"
}
```
